src/services/productService.js

```diff
@@ -274,5 +274,34 @@ class ProductServices {
       ? response
       : { response, message: "no match found" };
   }
+
+  static async markAvailable(id) {
+    const product = await Product.findOne({ where: { id } });
+    if (product.available === false) {
+      await Product.update({ available: true }, { where: { id } });
+      return { name: product.name, note: "Available!" };
+    }
+    if (product.available === true) {
+      await Product.update({ available: false }, { where: { id } });
+      return { name: product.name, note: "Not available!" };
+    }
+  }
+
+  static async updateItem(req) {
+    const product = await Product.findOne({ where: { id: req.params.id } });
+    if (!req.body.expiryDate) {
+      req.body.expiryDate = product.expiryDate;
+    }
+    const { price, name, bonus, expiryDate, categoryId, quantity } = req.body;
+    const images = req.files.map((img) => img.path);
+    await Product.update(
+      { price, name, bonus, expiryDate, categoryId, quantity, images },
+      { where: { id: req.params.id } },
+    );
+  }
+
+  static async deleteItem(id) {
+    await Product.destroy({ where: { id } });
+  }
```

Comment:
@suave990  The code does not include any error handling for cases where the database operations fail. Adding appropriate error handling can make the code more robust and prevent unexpected crashes.